### ISSUE TITLE
Color changes light control stays white

### ISSUE THREAD

**Issue by Turtle-Toast** [mjwwit/node-red-contrib-ikea-tradfri#6]
I am trying to control the lights so the color changes to red. Yet everything time I try changing the color, it becomes white (or stays white).

I tried using an inject node:
```
Topic (Number): Bulb ID
Payload (JSON): {"color": "#ff0000", "brightness": 75}
```
The brightness or setting it on/off works fine, only the color does not change or it changes to white when I was a different color.

Using the light control node itself did not worked as well. The brightness or setting the bulb on/off worked fine, setting the color did not worked. The bulb is correctly defined as an RGB bulb.

Any ideas what I could try?

**Comment by mjwwit** [comment]
Thanks for reporting this! Sadly I don't have an RGB bulb at this moment to test with... Nevertheless I'll double check everything related to bulb color to see if I can spot potential issues. I'll let you know if I release a version you can test.

**Comment by Turtle-Toast** [comment]
> Thanks for reporting this! Sadly I don't have an RGB bulb at this moment to test with... Nevertheless I'll double check everything related to bulb color to see if I can spot potential issues. I'll let you know if I release a version you can test.

Thank you. I hope there is a simpel solution to this.
I use Node Red within Home Assistant. I used the palette to add this. I don't know if there is a problem due that the Node Red site says it is at version 0.4.1 and this is 0.4.3. ([Node Red Site](https://flows.nodered.org/node/node-red-contrib-ikea-tradfri))

**Comment by nowi967** [comment]
Hi, I have the same issue.
I have NodeRed running under ioBroker.
My lights can be dimmed and switched on and off.
But the tuneble white "colorTemperature": does not work.

**Comment by mjwwit** [comment]
> But the tuneble white "colorTemperature": does not work.
That's odd, it works fine for me. What does the message you send to the `tradfri-light-control` node look like?
Mine looks like this:

```json
{
  "topic": [1234, 4321],
  "payload": {
    "onOff": true,
    "transitionTime": 0.5,
    "colorTemperature": 50
  }
}
```

**Comment by nowi967** [comment]
<img width="555" alt="Bildschirmfoto 2021-05-04 um 16 54 17" src="https://user-images.githubusercontent.com/67169020/117023801-b943e200-acf9-11eb-89eb-ac01d4f44e44.png">
I set the settings directly in the node. Like in the screenshot.

**Comment by sttts** [comment]
Same here. Color changes from nodered to tradfri are broken.

**Comment by mjwwit** [comment]
> <img alt="Bildschirmfoto 2021-05-04 um 16 54 17" width="555" src="https://user-images.githubusercontent.com/67169020/117023801-b943e200-acf9-11eb-89eb-ac01d4f44e44.png">
> 
> I set the settings directly in the node. Like in the screenshot.

Can you try it without setting brightness? I think I remember there being some settings that don't play nice together.

**Comment by Turtle-Toast** [comment]
When I use an Inject Node to send only `{"color":"#ff0000"}` (and a topic), the light simply turns off, when turned on, the color is white again (Changed from a warmer 'summer white' to cold 'hospital white' color). That it changes to white also happens when I use the Light Control Node itself and only set the color/bulb. I tried setting the bulb off (manually) and change color and setting the bulb on (manually) and change color.

**Comment by mjwwit** [comment]
I've just released a new version (0.4.4) that may fix your issues with setting color or colorTemperature on lights. It may take a while for your Node-RED palette to notice the update, but you can probably already manually install it using npm.
I've tested the fix on my color temperature bulbs, and changing brightness or colorTemperature (or a combination of the two) works perfectly now.
Can someone test this for rgb bulbs?

**Comment by sttts** [comment]
I don't see a difference.

**Comment by mjwwit** [comment]
> * setting the dimmer switches off the light.

I missed this one while testing, it appears to suffer from the same issue as `color` and `colorTemperature`. I will release a fix for this asap.
> * color changes from Homee have no effect.

Homee?
> * switching on/off sets the color back to some white tone.

The color temperature bulbs I use do not seem to have this issue. I can turn them off and back on and they'll remember the previously set temperature. I can even cut the power to the bulbs and it will still remember its settings.
Sadly I do not own any RGB color bulbs, so I cannot test any changes I make. If anyone's willing to put in some elbow grease I'd really love a PR for this.

**Comment by sttts** [comment]
> Sadly I do not own any RGB color bulbs

How can we make you own one? :-)

**Comment by mjwwit** [comment]
I'll buy one next time I'm at IKEA. That might take a while though, due to pandemic.

**Comment by sttts** [comment]
Yes, sending to a light, not group. 
```
msg.payload = { 
            color: "#123456",
};
```

**Comment by mjwwit** [comment]
Can you try using the `hue` and `saturation` properties?

**Comment by sttts** [comment]
Doesn't seem to help. I see not color change at all. I think it is worth to check the underlying node library. I read something in their issue list about mis-recognized devices. Maybe this is such a case.

**Comment by tjibbechris** [comment]
I found out that supplying msg.payload.colorTemperature to the tradfri-light-control does change the temperature. 
Looks like the node sets ``temperature`` in stead of ``colorTemperature``.

![afbeelding](https://user-images.githubusercontent.com/11209915/119564725-68c81d80-bda9-11eb-968c-3e392446fd21.png)
vs 
![afbeelding](https://user-images.githubusercontent.com/11209915/119564781-741b4900-bda9-11eb-866f-4d92ec238161.png)

**Comment by mjwwit** [comment]
> I found out that supplying msg.payload.colorTemperature to the tradfri-light-control does change the temperature.
> Looks like the node sets `temperature` in stead of `colorTemperature`.

Good find! It seems I indeed made an error here. I'm not sure if this fixes any of the original problems, but I've fixed this and released version 0.4.6.

**Comment by mjwwit** [comment]
I was finally able to go to IKEA and buy an RGB bulb last weekend. After some careful debugging and testing I've found and fixed the underlying problem leading to white RGB bulbs and bulbs not remembering their color when turned on.
I've already released version 0.4.8 which includes this fix.
